target_chains/ethereum/entropy_sdk/solidity/PRNG.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+/// @title PRNG Library
+/// @notice A library for pseudorandom number generation and related utility functions
+/// @dev This PRNG library is designed to work with Pyth Entropy as the seed source.
+///      Pyth Entropy provides secure, rapid random number generation for blockchain applications,
+///      enabling responsive UX for NFT mints, games, and other use cases requiring randomness.
+library PRNG {
+    /// @notice Generate a random byte between 0 and 255
+    /// @param seed The Pyth Entropy seed (bytes32)
+    /// @return A random uint8 value
+    function randomByte(bytes32 seed) internal pure returns (uint8) {
+        return uint8(uint256(keccak256(abi.encode(seed))) % 256);
+    }
+
+    /// @notice Generate a sequence of random bytes
+    /// @param seed The Pyth Entropy seed (bytes32)
+    /// @param length The number of random bytes to generate (max 32)
+    /// @return A bytes array of random values
+    function randomBytes(
+        bytes32 seed,
+        uint256 length
+    ) internal pure returns (bytes memory) {
+        require(length <= 32, "Length must be 32 or less");
+        bytes32 randomness = keccak256(abi.encode(seed));
+        bytes memory result = new bytes(length);
+        assembly {
+            mstore(add(result, 32), randomness)
+        }
+        return result;
+    }
+
+    /// @notice Generate a random integer within a specified range
+    /// @param seed The Pyth Entropy seed (bytes32)
+    /// @param min The minimum value (inclusive)
+    /// @param max The maximum value (inclusive)
+    /// @return A random uint256 value between min and max
+    function randInt(
+        bytes32 seed,
+        uint256 min,
+        uint256 max
+    ) internal pure returns (uint256) {
+        require(max > min, "Max must be greater than min");
+        return (uint256(keccak256(abi.encode(seed))) % (max - min + 1)) + min;
+    }
+
+    /// @notice Expand a seed into an array of random bytes32 values
+    /// @param seed The Pyth Entropy seed (bytes32)
+    /// @param numHashes The number of random bytes32 values to generate
+    /// @return An array of random bytes32 values
+    function expandRandomness(
```

Comment:
(if the PRNG had internal state, then this method could be replaced by the user simply calling `randBytes32` repeatedly)